### ISSUE TITLE
Lint/AssignmentInCondition not triggered like I expected

### ISSUE THREAD

**Issue by issei126** [bbatsov/rubocop#3323]
I wanted to write a method to return the result of arguments value comparison boolean value.
But I missed and assignment to arguments in condition.

Rubocop's lint does not warn it. Is that exactly behavior ? 

--------

## Expected behavior

```
def foo(_bar, _hoo)
  _bar = 1 && _hoo = 2
end
```
Should trigger the `Lint/AssignmentInCondition`

## Actual behavior

```
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## Steps to reproduce the problem

```
$ cat test.rb
def foo(_bar, _hoo)
  _bar = 1 && _hoo = 2
end
$ rubocop test.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## RuboCop version

```
$ rubocop -V
0.41.2 (using Parser 2.3.1.2, running on ruby 2.2.1 x86_64-darwin14)
```

**Comment by rrosenblum** [comment]
`Lint/AssignmentInCondition` only looks for assignment inside of `if`, `unless`, `while`, and `until`. `if _bar = 1 && _hoo = 2` would register an offense.

I am not sure I fully understand what the example code is attempting to accomplish. Maybe there is room for another check.

**Comment by issei126** [comment]
I'm sorry I have tell my idea well.

The method I wanted to write is
```
def foo(_bar, _hoo)
  _bar == 1 && _hoo == 2
end
```
The method will compare to argument values and return the result.
But the example code return always true. It is not expected method behavior.
So I want Rubocop to trigger 'Lint/AssignmentInCondition'

**Comment by jonas054** [comment]
Sure, it makes sense to extend the responsibilities of `Lint/AssignmentInCondition` to also check expressions that are obviously conditions, such as return values of predicate methods.

I notice, however, that you had to prepend an underscore to the parameter names in order to avoid reports from two other cops, `Lint/UnusedMethodArgument` and `Lint/UselessAssignment`, so even without your proposed changes the unintended assignments don't go unnoticed in normal cases.

It's not trivial to detect all expressions that are possible return values, but I guess we can reuse code from other cops. Anyway, this change should not be a top priority as `Lint/UselessAssignment` steps up in all normal cases, as far as I can see.

**Comment by issei126** [comment]
Thank you for your reply.

I see that.  I try to detect miss by `Lint/UnusedMethodArgument` and `Lint/UselessAssignment` cops

I close this issue.
